File: DarkOrbit 10.0/Net/netty/commands/AttackTypeModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class AttackTypeModule
    {
        public const short ID = 32205;

        public const short ROCKET = 0;
        public const short LASER = 1;
        public const short MINE = 2;
        public const short RADIATION = 3;
        public const short PLASMA = 4;
        public const short ECI = 5;
        public const short SL = 6;
        public const short CID = 7;
        public const short SINGULARITY = 8;
        public const short KAMIKAZE = 9;
        public const short REPAIR = 10;
        public const short DECELERATION = 11;
        public const short SHIELD_ABSORBER_ROCKET_CREDITS = 12;
        public const short SHIELD_ABSORBER_ROCKET_URIDIUM = 13;
        public const short STICKY_BOMB = 14;
        public const short varQ3U = 15;

        public static short typeValue = 0;

        public AttackTypeModule(short param1)
        {
            typeValue = param1;
        }

        public byte[] write()
        {
            ByteArray param1 = new ByteArray(ID);
            param1.writeShort(typeValue);
            param1.writeShort(-6647);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/AttackTypeModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

/**
 Created by Pedro on 25-02-2015.
 */
public class AttackTypeModule {

    public static short ROCKET       = 0;
    public static short REPAIR       = 10;
    public static short const_1488   = 15;
    public static short SINGULARITY  = 8;
    public static short DECELERATION = 11;
    public static short CID          = 7;
    public static short KAMIKAZE     = 9;
    public static short STICKY_BOMB  = 14;
    public static short PLASMA       = 4;
    public static short MINE         = 2;
    public static short const_383    = 13;
    public static short RADIATION    = 3;
    public static short SL           = 6;
    public static short ECI          = 5;
    public static short LASER        = 1;
    public static short const_236    = 12;
    public static int   ID           = 14619;
    public        short var_305      = 0;

    public AttackTypeModule(short param1) {
        this.var_305 = param1;
    }


    public void write(DataOutputStream pDataout) {
        try {
            pDataout.writeShort(ID);
            this.writeInternal(pDataout);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream pDataout) {
        try {
            pDataout.writeShort(-20422);
            pDataout.writeShort(this.var_305);
        } catch (IOException e) {
        }
    }
}
